test/test_shmem.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2024 Ventana Micro Systems Inc.
+ */
+
+#include <librpmi.h>
+
+#undef DEBUG
+
+#ifdef DEBUG
+#define DPRINTF(msg...)         rpmi_env_printf(msg)
+#else
+#define DPRINTF(msg...)
+#endif
+
+struct rpmi_shmem *rpmi_shmem_qemu_create(const char *name, rpmi_uint64_t base,
+					  rpmi_uint32_t size);
+/** RPMI shared memory instance */
+struct rpmi_shmem {
+	/** Name of the shared memory */
+	const char *name;
+	/** Base address of the shared memory */
+	rpmi_uint64_t base;
+	/** Size of the shared memory */
+	rpmi_uint32_t size;
+	/** Pointer to platform specific operations */
+	const struct rpmi_shmem_platform_ops *ops;
+	/** Private data of the platform specific oprations */
+	void *ops_priv;
+};
+
+static enum rpmi_error shmem_env_memcpy_read(void *priv, rpmi_uint64_t addr,
+					     void *in, rpmi_uint32_t len)
+{
+	rpmi_env_memcpy(in, (void *)addr, len);
```

Comment:
Never directly case integer to pointer. Instead first cast it to `unsigned long` and then cast it to pointer.

For example: `rpmi_env_memcpy(in, (void *)(unsigned long)addr, len);`

Reply:
This is to avoid compilation errors for 32bit systems.

Reply:
This file is gone now.